### PR TITLE
HtmlEditor: first selected cell's color should be passed as a colorBox default when editing several cells color (T1127428)

### PR DESCRIPTION
**Pull request by alexanderPolosatov**

### CODE REVIEW

**Review by Zedwag — DISMISSED** [review]

**Comment by ksercs** [comment]
@let's make the PR's name more concrete, for example, "HtmlEditor: first selected cell's color should be passed as a colorBox default when editing several cells color (T1127428)"

File: js/ui/html_editor/utils/toolbar_helper.js [hunk 1]
```diff
@@ -184,6 +184,10 @@ function getLinkRange(module, range) {
     return result;
 }
 
+function processColorFromFormats(value) {
```
Comment (ksercs) [inline js/ui/html_editor/utils/toolbar_helper.js]:
```suggestion
function getColorFromFormat(value) {
```

File: testing/tests/DevExpress.ui.widgets.htmlEditor/htmlEditorParts/tableProperties.tests.js [hunk 2]
```diff
@@ -210,6 +218,40 @@ module('Table properties forms', {
             assert.strictEqual($tableElement.css('textAlign'), 'right', 'text align is applied');
         });
 
+        test('Passing backgroundColor & borderColor from empty selected cells to colorBox', function(assert) {
```
Comment (ksercs) [inline testing/tests/DevExpress.ui.widgets.htmlEditor/htmlEditorParts/tableProperties.tests.js]: Let's answer the question "how should it work?"
For example, ""
```suggestion
        test('cell backgroundColor should be passed to colorBox as a default during editing', function(assert) {
```

File: testing/tests/DevExpress.ui.widgets.htmlEditor/htmlEditorParts/tableProperties.tests.js [hunk 3]
```diff
@@ -210,6 +218,40 @@ module('Table properties forms', {
...
+
+            this.quillInstance.setSelection(0, 2);
+            showCellPropertiesForm(this.instance, $tableElement);
+            this.clock.tick();
+
+            const formInstance = this.getFormInstance();
+            const backgroundColorEditor = formInstance.$element().find('.dx-colorbox').eq(1).dxColorBox('instance');
+            const borderColorEditor = formInstance.$element().find('.dx-colorbox').eq(0).dxColorBox('instance');
+
+            assert.strictEqual(backgroundColorEditor.option('value'), 'green', 'background color passed to colorBox');
```
Comment (ksercs) [inline testing/tests/DevExpress.ui.widgets.htmlEditor/htmlEditorParts/tableProperties.tests.js]:
```suggestion
            assert.strictEqual(backgroundColorEditor.option('value'), 'green', 'background color is passed to colorBox');
```

File: testing/tests/DevExpress.ui.widgets.htmlEditor/htmlEditorParts/tableProperties.tests.js [hunk 4]
```diff
@@ -210,6 +218,40 @@ module('Table properties forms', {
...
+            assert.strictEqual(borderColorEditor.option('value'), 'gray', 'border color passed to colorBox');
```
Comment (ksercs) [inline testing/tests/DevExpress.ui.widgets.htmlEditor/htmlEditorParts/tableProperties.tests.js]: Let's separate this cases to 2 different tests? We can reduce copypaste using a separate module (or maybe forEach)

File: testing/tests/DevExpress.ui.widgets.htmlEditor/htmlEditorParts/tableProperties.tests.js [hunk 5]
```diff
@@ -210,6 +218,40 @@ module('Table properties forms', {
...
+        });
+
+        test('Apply backgroundColor & borderColor of first selected call for all cells when color did not modified in colorBox', function(assert) {
```
Comment (ksercs) [inline testing/tests/DevExpress.ui.widgets.htmlEditor/htmlEditorParts/tableProperties.tests.js]:
```suggestion
        test('backgroundColor of first selected cell should be applied for all cells when color was not modified in colorBox', function(assert) {
```

File: testing/tests/DevExpress.ui.widgets.htmlEditor/htmlEditorParts/tableProperties.tests.js [hunk 6]
```diff
@@ -210,6 +218,40 @@ module('Table properties forms', {
...
+            this.clock.tick();
+
+            this.applyFormChanges();
+
+            assert.strictEqual($tableElement.find('td').eq(0).css('backgroundColor'), 'rgb(0, 128, 0)', 'background color is applied');
+            assert.strictEqual($tableElement.find('td').eq(0).css('borderColor'), 'rgb(128, 128, 128)', 'borderColor color is applied');
+            assert.strictEqual($tableElement.find('td').eq(1).css('backgroundColor'), 'rgb(0, 128, 0)', 'background color is applied');
+            assert.strictEqual($tableElement.find('td').eq(1).css('borderColor'), 'rgb(128, 128, 128)', 'borderColor color is applied');
+        });
+
```
Comment (ksercs) [inline testing/tests/DevExpress.ui.widgets.htmlEditor/htmlEditorParts/tableProperties.tests.js:255]: don't we want test that first selected cell's color is passed to colorBox during editing?

File: testing/tests/DevExpress.ui.widgets.htmlEditor/htmlEditorParts/tableProperties.tests.js [hunk 7]
```diff
@@ -210,6 +218,40 @@ module('Table properties forms', {
...
+            this.createWidget({ value: tableWithoutContent });
+
+            const $tableElement = this.$element.find('table').eq(0);
+
+            this.quillInstance.setSelection(0, 2);
+            showCellPropertiesForm(this.instance, $tableElement);
+            this.clock.tick();
+
+            const formInstance = this.getFormInstance();
+            const backgroundColorEditor = formInstance.$element().find('.dx-colorbox').eq(1).dxColorBox('instance');
```
Comment (EugeniyKiyashko) [inline testing/tests/DevExpress.ui.widgets.htmlEditor/htmlEditorParts/tableProperties.tests.js]: Let's extract class names into separate const variables

File: testing/tests/DevExpress.ui.widgets.htmlEditor/htmlEditorParts/tableProperties.tests.js (hunk 6 above)
Comment (alexanderPolosatov) [reply testing/tests/DevExpress.ui.widgets.htmlEditor/htmlEditorParts/tableProperties.tests.js:255]: it's checked [here](https://github.com/DevExpress/DevExtreme/pull/23825/files#diff-d6397b99967eae72a6fe372a0d6477ffdb29c874322016ce4e2e3318fd4766f5R221)

File: testing/tests/DevExpress.ui.widgets.htmlEditor/htmlEditorParts/tableProperties.tests.js (hunk 4 above)
Comment (alexanderPolosatov) [reply testing/tests/DevExpress.ui.widgets.htmlEditor/htmlEditorParts/tableProperties.tests.js]: I will replace to forEach

**Review by ksercs — APPROVED** [review]

**Review by EugeniyKiyashko — APPROVED** [review]

**Pull request merged**